packages/signers/meta-signers/src/multisig-signer.ts

```diff
@@ -0,0 +1,199 @@
+        if (tmpMeta.tag !== "v14" && tmpMeta.tag !== "v15") throw null
+        lookup = getLookupFn(tmpMeta.value)
+        if (lookup.call === null) throw null
+        dynamicBuilder = getDynamicBuilder(lookup)
+      } catch (_) {
+        throw new Error("Unsupported metadata version")
+      }
+
+      // Try as_multi_threshold_1
+      if (multisig.threshold === 1) {
+        try {
+          const { location, codec } = dynamicBuilder.buildCall(
+            "Multisig",
+            "as_multi_threshold_1",
+          )
+          const payload = codec.enc({
+            other_signatories: otherSignatories.map(toSS58),
+            call: dynamicBuilder.buildDefinition(lookup.call).dec(callData),
+          })
+          const wrappedCallData = mergeUint8(new Uint8Array(location), payload)
+          return signer.signTx(
+            wrappedCallData,
+            signedExtensions,
+            metadata,
+            atBlockNumber,
+            hasher,
+          )
+        } catch (_) {}
+      }
+
+      const unsignedExtrinsic = mergeUint8(new Uint8Array([4]), callData)
+      const [multisigInfo, weightInfo] = await Promise.all([
+        getMultisigInfo(toSS58(multisigId), Binary.fromBytes(callHash)),
+        txPaymentInfo(
+          Binary.fromBytes(unsignedExtrinsic),
+          unsignedExtrinsic.length,
+        ),
+      ])
+
+      if (
+        multisigInfo?.approvals.some((approval) => {
+          const info = getSs58AddressInfo(approval)
+          return info.isValid && u8ArrEq(info.publicKey, signer.publicKey)
+        })
+      ) {
+        throw new Error("Multisig call already approved by signer")
+      }
+
+      const method = options.method(
+        multisigInfo?.approvals ?? [],
+        multisig.threshold,
+      )
+
+      let wrappedCallData
+      try {
+        const { location, codec } = dynamicBuilder.buildCall("Multisig", method)
+        const payload = codec.enc({
+          threshold: multisig.threshold,
+          other_signatories: otherSignatories.map(toSS58),
+          max_weight: weightInfo.weight,
+          maybe_timepoint: multisigInfo?.when,
+          ...(method === "as_multi"
+            ? {
+                call: dynamicBuilder.buildDefinition(lookup.call).dec(callData),
+              }
+            : {
+                call_hash: Binary.fromBytes(callHash),
+              }),
+        })
+        wrappedCallData = mergeUint8(new Uint8Array(location), payload)
+      } catch (_) {
+        throw new Error(
+          `Unsupported runtime version: Multisig.${method} not present or changed substantially`,
+        )
+      }
+
+      return signer.signTx(
+        wrappedCallData,
+        signedExtensions,
+        metadata,
+        atBlockNumber,
+        hasher,
+      )
+    },
+  }
+}
+
+const u8ArrEq = (a: Uint8Array, b: Uint8Array) => {
+  if (a.length != b.length) return false
+  return Array.from(a).every((v, i) => v === b[i])
```

Comment:
Is there any reason to do `Array.from`? [`TypedArray`s should have as well `every`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/every).

Reply:
Yeah... fear of unsupported feature 😅 I see that browsers implemented this a decade ago, so it should™ be fine